src/components/svgImage/index.tsx

```diff
@@ -28,7 +30,12 @@ function SvgImage(props: SvgImageProps) {
   if (isSvgUri(data)) {
     return <SvgCssUri {...others} uri={data.uri}/>;
   } else if (typeof data === 'string') {
-    return <SvgXml xml={data} {...others}/>;
+    const flattenStyle = StyleSheet.flatten(props.style) as Record<string, any>;
```

Comment:
We use `flatten` very rarely, why is this required here?

Reply:
to extract the `width` and `height` from the provided styles.
do you have suggestion how to do it differently?

Reply:
are we sure its coming with more than one style object ?
if not, and we always have just single style object (and not as array).?
we can remove that.

Reply:
yes i'm sure. we can't remove it

Reply:
Can you please provide the scenario this is used in?

Reply:
Yes, for example when using Button with SVG Icon, we're passing `iconStyle` with `width` & `height`.
in this case it's been passed to the SVG component, and without this PR the height and width is not respected.

Reply:
```
    <Button
      id={id}
      testID={TestIds.buttonElement}
      style={style}
      color={color}
      iconSource={iconSource}
      iconStyle={{width: Spacings.s6, height: Spacings.s6}}
      round
      size={Button.sizes.large}
    />
```

Reply:
As discussed, lets allow passing `iconProps` to the `Button` instead, which will allow sending `width` and `height`